.github/workflows/main.yml

```diff
@@ -19,25 +19,77 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      # TODO: cache all the setup
-
       - name: Setup directories
         run: |
           # create directory that will be on the PATH
           mkdir -p ~/.local/bin
           echo "${HOME}/.local/bin" >> $GITHUB_PATH
           mkdir -p ~/source
+     
+      - uses: actions/cache@v2
+        id: cache-gcc
+        with:
+          path: ~/.local/bin/gcc
```

Comment:
The only thing this script does in `~/.local/bin` is set up some links that take a negligible amount of time to set up.  I don't think you need to cache `~/.local/bin/gcc` - figure out where GCC is actually installed to and use that directory instead.